File: apps/web/core/constants/initiative.ts
/**
 * SPDX-FileCopyrightText: 2023-present Plane Software, Inc.
 * SPDX-License-Identifier: LicenseRef-Plane-Commercial
 *
 * Licensed under the Plane Commercial License (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * https://plane.so/legals/eula
 *
 * DO NOT remove or modify this notice.
 * NOTICE: Proprietary and confidential. Unauthorized use or distribution is prohibited.
 */

import { INITIATIVE_STATES } from "@plane/constants";
import type {
  TCreateUpdateInitiativeModal,
  TBaseLayoutType,
  TInitiativeGroupByOptions,
  TInitiativeOrderByOptions,
} from "@plane/types";

export const DEFAULT_CREATE_UPDATE_INITIATIVE_MODAL_DATA: TCreateUpdateInitiativeModal = {
  isOpen: false,
  initiativeId: undefined,
};

export const INITIATIVE_GROUP_BY_OPTIONS: {
  key: TInitiativeGroupByOptions;
  title: string;
}[] = [
  { key: "lead", title: "Lead" },
  { key: "created_by", title: "Created By" },
  { key: "state", title: "States" },
  { key: "label_ids", title: "Labels" },
  { key: undefined, title: "None" },
];

export const INITIATIVE_ORDER_BY_OPTIONS: {
  key: TInitiativeOrderByOptions;
  title: string;
}[] = [
  { key: "-created_at", title: "Last Created" },
  { key: "-updated_at", title: "Last Updated" },
];

export const DEFAULT_INITIATIVE_STATE = INITIATIVE_STATES.DRAFT.key;
export const DEFAULT_INITIATIVE_LAYOUT: TBaseLayoutType = "list";

File: apps/api/plane/api/urls/initiative.py
# SPDX-FileCopyrightText: 2023-present Plane Software, Inc.
# SPDX-License-Identifier: LicenseRef-Plane-Commercial
#
# Licensed under the Plane Commercial License (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
# https://plane.so/legals/eula
#
# DO NOT remove or modify this notice.
# NOTICE: Proprietary and confidential. Unauthorized use or distribution is prohibited.

from plane.api.views import InitiativeViewSet, InitiativeLabelViewSet, InitiativeEpicsViewSet, InitiativeProjectsViewSet
from django.urls import path


urlpatterns = [
    # Initiative urls - manage initiatives, and its associated labels, projects, and epics
    path(
        "workspaces/<str:slug>/initiatives/",
        InitiativeViewSet.as_view({"get": "list", "post": "create"}),
        name="initiatives",
    ),
    path(
        "workspaces/<str:slug>/initiatives/<uuid:pk>/",
        InitiativeViewSet.as_view({"get": "retrieve", "patch": "partial_update", "delete": "destroy"}),
        name="initiatives",
    ),
    path(
        "workspaces/<str:slug>/initiatives/<uuid:initiative_id>/labels/",
        InitiativeViewSet.as_view({"get": "get_labels", "post": "add_labels", "delete": "remove_labels"}),
        name="initiative-labels-manage",
    ),
    path(
        "workspaces/<str:slug>/initiatives/<uuid:initiative_id>/projects/",
        InitiativeProjectsViewSet.as_view({"get": "get_projects", "post": "add_projects", "delete": "remove_projects"}),
        name="initiative-projects-manage",
    ),
    path(
        "workspaces/<str:slug>/initiatives/<uuid:initiative_id>/epics/",
        InitiativeEpicsViewSet.as_view({"get": "get_epics", "post": "add_epics", "delete": "remove_epics"}),
        name="initiative-epics-manage",
    ),
    # initiative labels endpoints
    path(
        "workspaces/<str:slug>/initiatives/labels/",
        InitiativeLabelViewSet.as_view({"get": "list", "post": "create"}),
        name="initiative-labels",
    ),
    path(
        "workspaces/<str:slug>/initiatives/labels/<uuid:pk>/",
        InitiativeLabelViewSet.as_view({"get": "retrieve", "patch": "partial_update", "delete": "destroy"}),
        name="initiative-label-detail",
    ),
]
